Sources/Apodini/Components/Component/Component.swift

```diff
@@ -33,10 +45,31 @@ extension Component {
     /// We require that each Component that conforms to `SyntaxTreeVisitable` provides its own custom `accept` implementation to avoid an endless loop in the `accept` function.
     public func accept(_ visitor: SyntaxTreeVisitor) {
         preconditionTypeIsStruct(Self.self, messagePrefix: "Component")
+
         if let visitable = self as? SyntaxTreeVisitable {
+            // This cases covers any Components conforming to SyntaxTreeVisitable.
+            // Most commonly this are Modifiers, but also Components like `Group`
+            // or special purpose Components like `TupleComponent`
+
+            // As stated above, this might be a Modifier and the Metadata of Modifiers can't be accessed.
+            // So we only start parsing the metadata if in fact we know that it isn't a Modifier.
+            if StandardModifierVisitor()(self) != true {
+                let metadata: Self.Metadata = self.metadata
+                (metadata as! AnyMetadata).accept(visitor)
```

Comment:
Why isn't the linter complaining here? I think you can directly go with `(metadata as? AnyMetadata)?.accept(visitor)` without the need of declaring a new variable on line 57

Reply:
Though it might make sense to make it explicit why `metadata.accept(...)` does not work, because Swift seems to think, that `Self.Metadata` is not a `AnyComponentOnlyMetadata` (see line 20). Not doing the cast would always result in calling `.accept` on the default Empty Metadata defined in line 33.
But I'm also fine merging it into one line, regression testing is in place.

Though not sure if I want to use optional chaining, after all it shouldn't be ignored silently if this fails.

Reply:
And regarding SwiftLint, has been really unreliable for me as well. Missed e.g. some spots where it should have complained with `missing_docs`.

Reply:
Resolved d59f0bc